satellite/gc/gc_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+	"storj.io/storj/uplink"
+)
+
+// TestGarbageCollection does the following:
+// * Upload two objects
+// * Put one storagenode offline
+// * Delete one object
+// * Put the storagenode back online
+// * Trigger a bloom filter generation
+// * Check that pieces of the deleted object are deleted on the storagenode
+// * Check that pieces of the kept object are not deleted on the storagenode
```

Comment:
How about a simpler approach:
- Have only 1 storage node in testplanet instead of 5.
- Upload 2 files.
- Delete the pointer of the first file from the satellite (without using the uplink), e.g. `planet.Satellites[0].Metainfo.Service.Delete(ctx, path)`. This way the pointer is gone from the satellite, but the piece is still on the storage node.
- Trigger a bloom filter generation.
- Check that pieces of the deleted object are deleted on the storagenode.
- Check that pieces of the kept object are not deleted on the storagenode.

Reply:
Done :)
Good idea, I didn't think about it